django/core/servers/basehttp.py

```diff
@@ -219,12 +219,19 @@ def handle_one_request(self):
         handler.run(self.server.get_app())
 
 
-def run(addr, port, wsgi_handler, ipv6=False, threading=False, server_cls=WSGIServer):
+def run(addr, port, wsgi_handler, ipv6=False, threading=False,
+        server_cls=WSGIServer, on_bind=None):
     server_address = (addr, port)
+    cls_dict = {}
+    if on_bind is not None:
+        def server_bind(self):
+            super(httpd_cls, self).server_bind()
```

Comment:
Wondering if this would be better as: 

```suggestion
            server_cls.server_bind(self)
```

🤔

(The `super()` call is relying on the definition of `https_cls` from below, which is a bit 🤯 maybe)